hack/bicep-types-radius/src/generator/src/cmd/generate.ts

```diff
@@ -242,113 +243,50 @@ async function findReadmePaths(specsPath: string) {
-  for (const namespace of sortBy(keys(byProvider), x => x.toLowerCase())) {
-    markdown += `## ${namespace}\n`;
-
-    const byResourceType = groupBy(byProvider[namespace], x => x.split('@')[0].toLowerCase());
-    for (const resourceType of sortBy(keys(byResourceType), x => x.toLowerCase())) {
-      markdown += `### ${resourceType}\n`;
-
-      for (const typeString of sortBy(byResourceType[resourceType], x => x.toLowerCase())) {
-        const version = typeString.split('@')[1];
-        const jsonPath = index.Resources[typeString].RelativePath;
-        const anchor = `resource-${typeString.replace(/[^a-zA-Z0-9-]/g, '').toLowerCase()}`
-
-        markdown += `* [${version}](${path.dirname(jsonPath)}/types.md#${anchor})\n`;
-      }
+  const typesPaths = await findRecursive(baseDir, filePath => {
+    return path.basename(filePath) === 'types.json';
+  });
 
-      markdown += '\n';
-    }
+  const typeFiles: TypeFile[] = [];
+  for (const typePath of typesPaths) {
+    const content = await readFile(typePath, { encoding: 'utf8' });
+    typeFiles.push({
+      relativePath: path.relative(baseDir, typePath),
+      types: readTypesJson(content),
+    });
   }
+  const indexContent = await buildIndex(typeFiles,  log => logOut(logger, log), {name: "radius", version: await getRadiusRelease(), isSingleton: false} as TypeSettings);
 
-  return markdown;
+  await writeFile(`${baseDir}/index.json`, writeIndexJson(indexContent));
+  await writeFile(`${baseDir}/index.md`, writeIndexMarkdown(indexContent));
 }
 
-async function buildIndex(logger: ILogger, baseDir: string): Promise<TypeIndex> {
-  const typeFiles = await findRecursive(baseDir, filePath => {
-    return path.basename(filePath) === 'types.json';
-  });
-  
-  const resourceTypes = new Set<string>();
-  const resourceFunctions = new Set<string>();
-  const resDictionary: Dictionary<TypeIndexEntry> = {};
-  const funcDictionary: Dictionary<Dictionary<TypeIndexEntry[]>> = {};
-
-  // Use a consistent sort order so that file system differences don't generate changes
-  for (const typeFilePath of orderBy(typeFiles, f => f.toLowerCase(), 'asc')) {
-    const content = await readFile(typeFilePath, { encoding: 'utf8' });
-
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const types = JSON.parse(content) as any[];
-    for (const type of types) {
-      const resourceType = type[TypeBaseKind.ResourceType];
-      if (resourceType) {
-        if (resourceTypes.has(resourceType.Name.toLowerCase())) {
-          logOut(logger, `WARNING: Found duplicate type "${resourceType.Name}"`);
-          continue;
-        }
-        resourceTypes.add(resourceType.Name.toLowerCase());
-  
-        resDictionary[resourceType.Name] = {
-          RelativePath: path.relative(baseDir, typeFilePath),
-          Index: types.indexOf(type),
-        };
-        
-        continue;
+function getRadiusRelease(): Promise<string> {
```

Comment:
Thinking about this more and this actually might not work. PRs should get merged in with the "latest" tag but when we test, we'd want to test against a PR-specific tag, not "latest". So this might end up generating the wrong version number. Will look into this more

Reply:
Leaving this for now. Will address in a follow up PR when we get to uploading to ACR